OPAL/ide/src/main/scala/org/opalj/ide/solver/IDEAnalysis.scala

```diff
@@ -0,0 +1,1115 @@
+/* BSD 2-Clause License - see OPAL/LICENSE for details. */
+package org.opalj
+package ide
+package solver
+
+import scala.collection
+import scala.collection.immutable
+import scala.collection.mutable
+
+import org.opalj.br.analyses.SomeProject
+import org.opalj.br.fpcf.FPCFAnalysis
+import org.opalj.fpcf.Entity
+import org.opalj.fpcf.EOptionP
+import org.opalj.fpcf.EPS
+import org.opalj.fpcf.InterimEUBP
+import org.opalj.fpcf.InterimPartialResult
+import org.opalj.fpcf.PartialResult
+import org.opalj.fpcf.ProperPropertyComputationResult
+import org.opalj.fpcf.PropertyKey
+import org.opalj.fpcf.Results
+import org.opalj.fpcf.SomeEOptionP
+import org.opalj.fpcf.SomeEPK
+import org.opalj.fpcf.SomeEPS
+import org.opalj.ide.integration.IDEPropertyMetaInformation
+import org.opalj.ide.integration.IDERawProperty
+import org.opalj.ide.integration.IDETargetCallablesProperty
+import org.opalj.ide.problem.AllTopEdgeFunction
+import org.opalj.ide.problem.EdgeFunction
+import org.opalj.ide.problem.EdgeFunctionResult
+import org.opalj.ide.problem.FinalEdgeFunction
+import org.opalj.ide.problem.FlowFunction
+import org.opalj.ide.problem.IDEFact
+import org.opalj.ide.problem.IdentityEdgeFunction
+import org.opalj.ide.problem.IDEProblem
+import org.opalj.ide.problem.IDEValue
+import org.opalj.ide.problem.InterimEdgeFunction
+import org.opalj.ide.util.Logging
+
+/**
+ * Basic solver for IDE problems. Uses the exhaustive algorithm that was presented in the original IDE paper from 1996
+ * as base. For an example problem have a look at `LinearConstantPropagationProblem` in the TAC module. For an example
+ * of interacting IDE problems have a look at `LCPOnFieldsProblem` and `LinearConstantPropagationProblemExtended`.
+ *
+ * @author Robin Körkemeier
+ */
+class IDEAnalysis[Fact <: IDEFact, Value <: IDEValue, Statement, Callable <: Entity](
```

Comment:
There are many complex methods in this class that should be properly documented, at least with a ScalaDoc on what they do.